pkg/didcomm/protocol/didexchange/persistence/connection.go

```diff
@@ -59,3 +74,77 @@ func computeHash(bytes []byte) (string, error) {
 	hash := h.Sum(bytes)
 	return fmt.Sprintf("%x", hash), nil
 }
+
+// SaveConnectionRecorder saved the connection record against the connection id  in the store
+func (c *ConnectionRecorder) SaveConnectionRecorder(msg *didexchange.Message) error {
+	invitation := &didexchange.Invitation{}
+	err := json.Unmarshal(msg.Msg.Payload, invitation)
+	if err != nil {
+		return fmt.Errorf("error while unmarshalling save connection record %s : ", err)
+	}
+	connRecord := &connectionRecord{ThreadID: msg.ThreadID, ServiceEndPoint: invitation.ServiceEndpoint,
+		RecKey: invitation.RecipientKeys, State: msg.State}
+	connBytes, err := json.Marshal(connRecord)
+	if err != nil {
+		return fmt.Errorf("error while marshalling save connection record %s : ", err)
+	}
+	err = c.store.Put(msg.ConnectionID, connBytes)
+	if err != nil {
+		return fmt.Errorf("failed to write to store: %s", err)
+	}
+	return nil
+}
+
+// SaveProtocolID saves the hash of the protocol id which is a did and thread id as an ID and connection id as value.
+func (c *ConnectionRecorder) SaveProtocolID(newDid string, msg *didexchange.Message) error {
+	if newDid == "" {
+		request := &didexchange.Request{}
```

Comment:
I'm bit skeptical about deriving(logic) the id's at persistence layer. IMO persistence layer should just deal with basic sanity check on the input and saving to db.

Reply:
done